src/midi.rs

```diff
@@ -83,54 +95,82 @@ fn get_ports(midi_in: &MidiInput, print: bool) -> Result<std::vec::Vec<midir::Mi
 pub struct MidiConnection {
     port: MidiInputPort,
     port_name: String,
-    midi_in: Option<MidiInput>
+    midi_in: Option<MidiInput>,
+}
+
+fn port_prompt(ports: &Vec<MidiInputPort>, re_ask: Option<()>) -> usize {
+    let mut choice: usize = 0;
+    if let None = re_ask {
+        println!("Please choose your midi input.");
+    }
+    match read_input().expect("Couldn't read input.").parse::<usize>() {
+        Ok(c) => {
+            if c < ports.len() {
+                println!("You have chosen {}", c);
+                choice = c;
+            } else {
+                println!("Please enter a valid number of a port.");
```

Comment:
```suggestion
                println!("{}: Not a valid port number", c);
```

Reply:
Let's keep it light, we can cut the "please".
Sounds too .... microsoft excel 😄

Reply:
sdadfgdsaf